### ISSUE TITLE
F8 conflicts with

### ISSUE THREAD

**Issue by pcgeek86** [PowerShell/vscode-powershell#331]
### System Details

- Operating system name and version: Windows 10 Anniversary Update
- VS Code version: 1.6.1
- PowerShell extension version: 0.7.2
- Output from `$PSVersionTable`:

```
Name                           Value
----                           -----
PSVersion                      5.1.14393.206
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.206
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Issue Description

The F8 keyboard shortcut, to run the "current" line in PowerShell scripts, conflicts with the default VSCode keyboard shortcut for F8, which points to the `Go to Next Error or Warning` command.

Cheers,
**Trevor Sullivan**
Docker Captain
Microsoft MVP: Cloud & Data Center Management
https://trevorsullivan.net
https://twitter.com/pcgeek86

**Comment by daviwil** [comment]
Hmmm, interesting, haven't seen that one before.  Does VS Code show you any warning about that or does  Run Selection just overshadow it?  F8 is kinda ingrained in PowerShell users' workflow so it might be hard to not use that keybinding by default.  Any ideas?

**Comment by pcgeek86** [comment]
@daviwil The current behavior, that I'm observing, is that the PowerShell F8 key binding is taking precedence over the VSCode default. I'm not sure what the best solution is here. I see both sides to this issue. 

- `F8` is familiar to PowerShell ISE users
- `F8` is familiar to VSCode users, for a specific, default behavior

In my opinion, breaking the default editor behavior is worse than asking users to change their habits, when moving to a new development tool. It's understandable that, coming from ISE to VSCode, you'll have some differences in behavior.

Hence, my recommendation would be to re-assess keyboard shortcuts for the VSCode PowerShell Extension, and find a layout that makes sense. I'm happy to hop on a call with you to hash this out in a document, and create a documented proposal.

**Comment by rkeithhill** [comment]
I agree with @pcgeek86.  We could use <kbd>Alt+/</kbd> instead.  That aligns with the F# `SendLine` command and it doesn't appear to conflict with any existing, core shortcuts.

**Comment by pcgeek86** [comment]
I like that idea, Keith.

**Comment by GHRoss** [comment]
I'm not too keen on this. F8 is my go to key for running highlighted text in PowerShell just like it is in the ISE. Although I wouldn't have a problem if this was configurable as it currently is within VS Code's preferences.

**Comment by daviwil** [comment]
Yeah, I think there's enough built-in expectation from PowerShell developers that F8 should execute the selection.  However, conflicting with a built-in keybinding isn't great, so I'm thinking about providing some kind of prompt the first time the user loads the extension to ask them if they want to use ISE-style keybindings and then auto-configure them if they confirm.  The defaults would be designed not conflict with existing VS Code bindings by default.

What do you think?

**Comment by GHRoss** [comment]
Love that idea. I'd go straight for the ISE bindings.

**Comment by rkeithhill** [comment]
Ditto except that I'd punt on the ISE bindings.  :-)  Seriously though, these days I'm using VS Code for other languages as well e.g.  C# and PHP.  It is nice when I can count on the VS Code shortcuts I know regardless of which extension is active.  This is one thing I didn't like about R# for a long time until they gave us the option to retain VS shortcuts during install.  Much like you are suggesting here.

**Comment by andyleejordan** [comment]
This is built in now. F8 (run selection) is the default keybinding, and the extension ships an **ISE Mode**: run **PowerShell: Enable ISE Mode** (or **Toggle ISE Mode**) to apply ISE-style keybindings and layout — there's even a "Try ISE mode" prompt in the walkthrough. Closing as implemented; please comment if something specific is still missing.

<sub>🤖 Drafted by GitHub Copilot at @andyleejordan's request — pending maintainer review.</sub>
